source/common/router/config_impl.cc

```diff
@@ -450,7 +450,7 @@ bool VirtualHostImpl::usesRuntime() const {
 VirtualHostImpl::VirtualClusterEntry::VirtualClusterEntry(
     const envoy::api::v2::VirtualCluster& virtual_cluster) {
   if (virtual_cluster.method() != envoy::api::v2::RequestMethod::METHOD_UNSPECIFIED) {
-    method_ = envoy::api::v2::RequestMethod_Name(virtual_cluster.method());
+    method_.value(envoy::api::v2::RequestMethod_Name(virtual_cluster.method()));
```

Comment:
Hmm... I've added tests for operator= and they pass even without the explicit operator=, so I guess you were right and it was invoking the copy constructor.